### ISSUE TITLE
I don't seem to be able to insert a null into a database column

### ISSUE THREAD

**Issue by alatabbi** [NLog/NLog#3380]
Hi!  
I don't seem to be able to insert a null into a database column, below is my code/configs
in columns Venue and DepartmentID,  when I set the value/DepartmentID  to null, NLog keeps inserting empty string

Am I missing something?

**NLog version**: (e.g. 4.6.3)

**Platform**:  .Net 4.5    

```c#
//////////////////////////////////
**NLog version**: (e.g. 4.6.3)
var logger = LogManager.GetLogger("FUT");
var dbTarget = ((AsyncTargetWrapper)LogManager.Configuration.FindTargetByName("FUTLog")).WrappedTarget as DatabaseTarget;

dbTarget.DBHost = "****";
dbTarget.DBDatabase = "****";
dbTarget.DBUserName = "****";
dbTarget.DBPassword = "****";

LogManager.ReconfigExistingLoggers();
LogEventInfo eventInfo = new LogEventInfo();
eventInfo.Properties["DepartmentName"] = "SomeDepartmentName";
eventInfo.Properties["DepartmentID"] = 777;
// eventInfo.Properties["Venue"] = "Venue";
logger .Info(eventInfo); 
//////////////////////////////////
```

**Current NLog config** (xml or C#, if relevant)
```xml
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" >

<targets async="true">
<target name="FUTLog"
                xsi:type="Database"
                dbProvider="System.Data.SqlClient"
                commandType="StoredProcedure"
                commandText="[dbo].[sp_InsertFeatureUsageTracking]">
<parameter name="@FeatureID" layout="${event-properties:item=FeatureID}" />
<parameter name="@CategoryID" layout="${event-properties:item=CategoryID}" />
<parameter name="@DepartmentID" layout="${event-properties:item=DepartmentID}" dbType="${db-null}" />

<parameter name="@Venue" layout="${when:when='${event-properties:item=Venue}' == '':inner=${db-null}:else=${event-properties:item=Venue}}" dbType="${db-null}" />

<parameter name="@EventTimestamp" layout="${date:universalTime=true}" />
</target>
</targets>

    <rules>
        <logger name="FUT"  levels="Info" minlevel="Info" writeTo="FUTLog" />
    </rules>
</nlog>

```

The expected result is to insert null into columns Venue and DepartmentID

**Comment by alatabbi** [comment]
I found a way to do it, basically adding 

``` parameterType="System.Object" ```

to the nullable  columns parameters does the trick.

Thanks.

**Comment by 304NotModified** [comment]
> Unfortunately ${db-null} isn't supported with ${when} for now in NLog 4.6.

FYI, this will be fixed in NLog 4.6.4
